### PR TITLE
[stable/hadoop]  Optionally deploy Apache Knox for external access to HDFS via webhdfs:// 

### PR DESCRIPTION
**Pull request by pfisterer**
**What this PR does / why we need it**:

This PR adds the ability to **optionally** deploy [Apache Knox](http://knox.apache.org/) (a REST API and Application Gateway for the Apache Hadoop Ecosystem) to the Hadoop cluster. 

This allows accessing HDFS resources externally (i.e. outside Kubernetes) using `webhdfs://` URLs 

Example:
```
you@yourlaptop$ helm install --set knox.enabled=true .
you@yourlaptop$ export HADOOP_USER_NAME=root 
you@yourlaptop$ export HDFS_URL=webhdfs://<knox-svc-ip:<knox-svc-ip>
you@yourlaptop$ ./hdfs dfs -mkdir $HDFS_URL/some_dir
you@yourlaptop$ ./hdfs dfs -ls $HDFS_URL/
```
Output:
```
Found 1 items
drwxr-xr-x   - root supergroup          0 2018-05-11 11:34 webhdfs://192.168.64.5:30531/some_dir
```

**Special notes for your reviewer**:

@danisla I hope this is a sensible enhancement of the chart. The config files are the default ones provided from Apache Knox.

### CODE REVIEW

File: stable/hadoop/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: The Apache Hadoop software library is a framework that allows for the distributed processing of large data sets across clusters of computers using simple programming models.
 name: hadoop
-version: 1.0.7
+version: 1.0.8
```
Comment (timstoop) [inline stable/hadoop/Chart.yaml]: You're adding a new feature, please bump the minor version.

File: stable/hadoop/templates/hadoop-configmap.yaml [hunk 2]
```diff
@@ -105,6 +105,11 @@ data:
     <?xml version="1.0"?>
     <?xml-stylesheet type="text/xsl" href="configuration.xsl"?>
     <configuration>
+      <property>
```
Comment (timstoop) [inline stable/hadoop/templates/hadoop-configmap.yaml]: I know nothing about hadoop, but shouldn't this property be conditional based on `knox.enabled`?

**Review by timstoop — CHANGES_REQUESTED** [review]

File: stable/hadoop/templates/hadoop-configmap.yaml (hunk 2 above)
Comment (pfisterer) [reply stable/hadoop/templates/hadoop-configmap.yaml]: It is enabled (by default, cf. [docs](https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/hdfs-default.xml)). So it does not alter the behaviour before this PR. I thought, it is safer and more transparent to explicitly enable it. 

However, I don't mind to make the inclusion of this based on `knox.enabled`. What do you prefer?

**Comment by pfisterer** [comment]
@timstoop I've updated the minor version.

File: stable/hadoop/templates/hadoop-configmap.yaml (hunk 2 above)
Comment (timstoop) [reply stable/hadoop/templates/hadoop-configmap.yaml]: I'd prefer for @danisla to make a decision about that, as he's the maintainer of the Chart.

**Comment by timstoop** [comment]
cc @danisla As maintainer, can you chime in please?

**Comment by pfisterer** [comment]
@timstoop / @danisla: I've updated the PR to enable WebHDFS explicitly only if `knox.enabled` is true. I agree with @timstoop that it should be included only if Apache Knox is enabled.

**Comment by pfisterer** [comment]
@danisla is there something I can do to support you in reviewing this PR?

**Comment by danisla** [comment]
@pfisterer this is a nice addition to the chart. I noticed the default knox service type is LoadBalancer, is there any default auth for knox since in most cases a LoadBalancer service provisions a public address.

**Comment by pfisterer** [comment]
@danisla I'm a bit busy currently - but I'll update it to include the default (LDAP-based) authentication mechanism. Is that ok?

**Pull request closed without merging**